src/workerd/io/actor-cache-test.c++

```diff
@@ -1442,76 +1418,122 @@ KJ_TEST("ActorCache read retry on flush containing only puts") {
   KJ_ASSERT(KJ_ASSERT_NONNULL(promise.wait(ws)) == "123");
 }
 
-KJ_TEST("ActorCache read hard fail") {
+// KJ_TEST("ActorCache read hard fail") {
+//   ActorCacheTest test;
+//   auto& ws = test.ws;
+//   auto& mockStorage = test.mockStorage;
+
+//   // Don't use expectCached() this time because we don't want eagerlyReportExceptions(), because
+//   // we actually expect an exception.
+//   auto promise = test.get("foo").get<kj::Promise<kj::Maybe<kj::String>>>();
+//   test.put("bar", "456");
+//   test.delete_("baz");
+
+//   // Expect the get, but don't resolve yet.
+//   auto mockGet = mockStorage->expectCall("get", ws)
+//       .withParams(CAPNP(key = "foo"));
+
+//   // Fail out the read with non-disconnect.
+//   kj::mv(mockGet).thenThrow(KJ_EXCEPTION(FAILED, "read failed"));
+
+//   // The read propagates the error.
+//   KJ_EXPECT_THROW_MESSAGE("read failed", promise.wait(ws));
+
+//   // The read is NOT retried, so expect the transaction to run now.
+//   auto mockTxn = mockStorage->expectCall("txn", ws).returnMock("transaction");
+//   mockTxn->expectCall("delete", ws)
+//       .withParams(CAPNP(keys = ["baz"]))
+//       .thenReturn(CAPNP(numDeleted = 0));
+//   mockTxn->expectCall("put", ws)
+//       .withParams(CAPNP(entries = [(key = "bar", value = "456")]))
+//       .thenReturn(CAPNP());
+//   mockTxn->expectCall("commit", ws).thenReturn(CAPNP());
+//   mockTxn->expectDropped(ws);
+
+//   // The read is NOT retried.
+//   mockStorage->expectNoActivity(ws);
+// }
+
+KJ_TEST("ActorCache read cancel") {
   ActorCacheTest test;
   auto& ws = test.ws;
   auto& mockStorage = test.mockStorage;
 
-  // Don't use expectCached() this time because we don't want eagerlyReportExceptions(), because
-  // we actually expect an exception.
-  auto promise = test.get("foo").get<kj::Promise<kj::Maybe<kj::String>>>();
+  (void)expectUncached(test.get("corge"));
```

Comment:
This has since been modified [in the other PR](https://github.com/cloudflare/workerd/pull/1916/commits/2531d9cd09115b4cff48a98c6af4dae15a970dc2) to test dropping before we yield the event loop, but as mentioned in another comment, that behavior looks a bit weird at the moment...